agent/agent.go

```diff
@@ -216,17 +216,20 @@ func (agent *Agent) healthCheck() {
 				}
 			}
 
-			traceUsage, ok := agent.metrics.Get("bytes_received_traces")
-			if !ok {
-				agent.logger.Errorf(context.Background(), "unexpected missing trace usage metric")
-			}
-			logUsage, ok := agent.metrics.Get("bytes_received_logs")
-			if !ok {
-				agent.logger.Errorf(context.Background(), "unexpected missing log usage metric")
-			}
+			recordUsage := agent.effectiveConfig.GetOpAMPConfig().RecordUsage
```

Comment:
You can do `agent.effectiveConfig.GetOpAMPConfig().RecordUsage.Get()`. Then you don't need to do the manual nil check and de-referencing the pointer

Reply:
sweet